src/vulkan/buffer.cc

```diff
@@ -71,6 +71,19 @@ void Buffer::CopyToDevice(VkCommandBuffer command) {
   MemoryBarrier(command);
 }
 
+void Buffer::CopyToHost(VkCommandBuffer command) {
+  if (is_buffer_host_accessible_)
```

Comment:
Even if the buffer is accessible to the host, you have to invalidate the mapped memory to ensure the host buffer gets the latest data.

https://www.khronos.org/registry/vulkan/specs/1.1/html/vkspec.html#vkInvalidateMappedMemoryRanges 

??